client/my-sites/plugins/plugin-saving/index.jsx

```diff
@@ -0,0 +1,44 @@
+import formatCurrency from '@automattic/format-currency';
+import { useSelector } from 'react-redux';
+import { getProductBySlug, isProductsListFetching } from 'calypso/state/products-list/selectors';
+
+export const PluginAnnualSaving = ( { plugin, children } ) => {
+	const priceSlugYearly = plugin?.variations?.yearly?.product_slug;
+	const productYearly = useSelector( ( state ) => getProductBySlug( state, priceSlugYearly ) );
+	const priceSlugMonthly = plugin?.variations?.monthly?.product_slug;
+	const productMonthly = useSelector( ( state ) => getProductBySlug( state, priceSlugMonthly ) );
+	const isFetching = useSelector( isProductsListFetching );
+
+	const getAnnualPriceSavingText = () => {
+		return productMonthly && productYearly
+			? formatCurrency(
+					Math.round( productMonthly.cost * 12 - productYearly.cost ),
+					productYearly.currency_code
+			  )
+			: null;
+	};
+
+	return children( {
+		isFetching,
+		saving: getAnnualPriceSavingText(),
+	} );
+};
+
+export const PluginAnnualSavingPercent = ( { plugin, children } ) => {
```

Comment:
Is this component used anywhere?

Reply:
Not yet, I prepared in case we need to use it in the future (all other saving texts, like in plans, use percent)

Reply:
One of [the values we follow for Calypso development](https://wpcalypso.wordpress.com/devdocs/docs/guide/0-values.md) I should have mentioned is iterating quickly and not writing code we don't need yet:

> 1. We optimize for iterating quickly
We aren’t trying to build software around a future we can’t predict. We don’t write code that we may not need. We know that no abstractions are better than bad abstractions.

That said, @SaxonF seems enthusiastic about having a percent option in here. 🙂 Is there one you prefer over the other Saxon?

Reply:
I removed the price saving percent component